src/EFCore.PG/Storage/ValueConversion/NpgsqlValueConverterSelector.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Storage.ValueConversion
+{
+    public class NpgsqlValueConverterSelector : ValueConverterSelector
+    {
+        readonly ConcurrentDictionary<(Type ModelElementClrType, Type ProviderElementClrType), ValueConverterInfo> _arrayConverters
+            = new ConcurrentDictionary<(Type, Type), ValueConverterInfo>();
+
+        public NpgsqlValueConverterSelector([NotNull] ValueConverterSelectorDependencies dependencies)
+            : base(dependencies) {}
+
+        /// <inheritdoc />
+        public override IEnumerable<ValueConverterInfo> Select(Type modelArrayClrType, Type providerArrayClrType = null)
+        {
+            var converters = default(IEnumerable<ValueConverterInfo>);
+            if (modelArrayClrType.IsArray && (providerArrayClrType == null || providerArrayClrType.IsArray))
```

Comment:
```csharp
providerArrayClrType == null || providerArrayClrType.IsArray
// equals to
providerArrayClrType?.IsArray != false
// because null != false and true != false
```

Reply:
Maybe, but I find the second option less clear than the first...